aten/src/ATen/native/Integration.cpp

```diff
@@ -0,0 +1,72 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/WrapDimUtils.h>
+#include <ATen/core/DimVector.h>
+#include <c10/util/Exception.h>
+
+namespace at {
+namespace native {
+namespace {
+
+// The estimated integral of a function y of x,
+// sampled at points (y_1, ..., y_n) that are separated by distance (dx_1, ..., dx_{n-1}),
+// is given by the trapezoid rule:
+//
+// \sum_{i=1}^{n-1}  dx_i * (y_i + y_{i+1}) / 2
+//
+// TODO: if we extend TensorIterator to accept 3 inputs,
```

Comment:
@colesbury new PR generalize the loops in TensorIterator to arbitrary number of inputs. But it hasn't been merged yet, and we would also need to implement the derivative by hand (not hard if considering only the point wise ops, but need to keep this in mind)

Reply:
Yeah, also I'd rather wait to see if this becomes an actual bottleneck for people before writing a kernel by hand. But I will keep it in mind as a possibility once Sam's PR lands.